File: packages/react-native/ios/BreezLiquidSDKEventListener.swift
import Foundation
import BreezLiquidSDK

class BreezLiquidSDKEventListener: EventListener {
    private var id: String?

    func setId(id: String) {
        self.id = id
        RNBreezLiquidSDK.addSupportedEvent(name: "event-\(id)")
    }

    func onEvent(e: LiquidSdkEvent) {
        if let id = self.id {
            if RNBreezLiquidSDK.hasListeners {
                RNBreezLiquidSDK.emitter.sendEvent(withName: "event-\(id)",
                                                   body: BreezLiquidSDKMapper.dictionaryOf(liquidSdkEvent: e))
            }
        }
    }
}

File: packages/react-native/android/src/main/java/com/breezliquidsdk/BreezLiquidSDKEventListener.kt
package com.breezliquidsdk

import breez_liquid_sdk.EventListener
import breez_liquid_sdk.LiquidSdkEvent
import com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter

class BreezLiquidSDKEventListener(
    private val emitter: RCTDeviceEventEmitter,
) : EventListener {
    private var id: String? = null

    fun setId(id: String) {
        this.id = id
    }

    override fun onEvent(e: LiquidSdkEvent) {
        this.id?.let {
            emitter.emit("event-$it", readableMapOf(e))
        }
    }
}
